trimesh/io/wavefront.py

```diff
@@ -193,23 +202,29 @@ def append_mesh():
                 loaded['metadata']['face_groups'] = face_groups
 
             if len(current['usemtl']) > 0:
-                current_mtllibs = [mtllibs[k] for k in current['usemtl']
-                                   if k in mtllibs]
-                try:
-                    vertex_colors = _get_vertex_colors(
-                        uv=loaded['metadata']['vertex_texture'],
-                        mtllibs=current_mtllibs,
-                        resolver=resolver)
-                    loaded['vertex_colors'] = vertex_colors
-                    loaded['metadata']['mtllibs'] = current_mtllibs
-                except BaseException:
-                    log.error('failed to load texture!', exc_info=True)
+                texture = np.array(current['vt'], dtype=np.float64)
+                vertex_colors = np.zeros((0, 4), dtype=np.uint8)
+                for usemtl, findices in usemtl_to_findices.items():
+                    uv = texture[findices]
+                    try:
+                        vertex_colors_i = _get_vertex_colors(
+                            uv=uv, mtllib=mtllibs[usemtl], resolver=resolver)
+                    except BaseException:
+                        log.error('failed to load texture: {}'.format(usemtl),
+                                  exc_info=True)
+                        vertex_colors_i = None
+                    if vertex_colors_i is None:
+                        vertex_colors_i = np.full(
+                            (len(uv), 4), (102, 102, 102, 255), dtype=np.uint8)
+                    vertex_colors = np.r_[vertex_colors, vertex_colors_i]
+                loaded['vertex_colors'] = vertex_colors[vert_order]
 
             # this mesh is done so append the loaded mesh kwarg dict
             meshes.append(loaded)
 
     attribs = {k: [] for k in ['v', 'vt', 'vn']}
     current = {k: [] for k in ['v', 'vt', 'vn', 'f', 'g', 'usemtl']}
+    usemtl_to_findices = collections.defaultdict(list)  # usemtl to 'f' indices
```

Comment:
This is necessary to get correspondence of usemtl and 'f v/vt/vn' after that, for example:

```
# xxx.obj
usemtl m0
f 0/0/0
f 1/1/1
...
usemtl m1
f 10/10/10
f 11/11/11
```